File: 1528-kids-with-the-greatest-number-of-candies/kids-with-the-greatest-number-of-candies.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        int maxcandy = 0;
        vector<bool> v;
        for (int i = 0; i < candies.size(); i++) {
            if (candies[i] > maxcandy) {
                maxcandy = candies[i];
            }
        }
        for (int i = 0; i < candies.size(); i++) {
            if ((candies[i] + extraCandies) >= maxcandy) {
                v.push_back(true);
            } else {
                v.push_back(false);
            }
        }
        return v;
    }
};

File: 1528-kids-with-the-greatest-number-of-candies/kids-with-the-greatest-number-of-candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        List<Boolean> result = new ArrayList<>();
        int largest = 0;
        for(int x : candies){
            if(x > largest){
                largest = x;
            }
        }

        for(int i = 0; i < candies.length; i++){
            if(candies[i] + extraCandies >= largest){
                result.add(true);
            }else{
                result.add(false);
            }
        }
    return result;
    }
}